packages/combobox/src/index.tsx

```diff
@@ -312,6 +312,8 @@ export const Combobox = React.forwardRef(function Combobox(
   let id = useId(props.id);
   let listboxId = id ? makeId("listbox", id) : "listbox";
 
+  let [isControlled, setIsControlled] = React.useState(false);
```

Comment:
Is there a reason we need new state for this? Could we not store this as a ref instead as we do lower in the tree? I don't believe we ever need it for rendering.

```ts
let isControlledRef = React.useRef(false);

// in ComboboxInput:
let isControlled = controlledValue != null;

React.useEffect(() => {
  isControlledRef.current = isControlled;
}, [isControlled]);

// In ComboboxOption:
let { isControlledRef } = React.useContext(ComboboxContext);
let handleClick = () => {
  onSelect && onSelect(value);
  transition(SELECT_WITH_CLICK, { value, isControlled: isControlledRef.current });
};
```

Reply:
@chaance that's exactly what I did originally, but decided to go with state, since setting that ref would not trigger a re-render, so I figured state would be more safe. 

It would almost certainly work either way though. If you think a ref would be better I could change to that.

Reply:
@chaance ok I've got it ref based now. I'm putting just the boolean in context, as opposed to the ref itself. If you'd like me to remove the updater function, and just store the ref object, lemme know and I will.